moonstone/filtering/filtering.py

```diff
@@ -0,0 +1,44 @@
+from moonstone.core.module_base import BaseModule
+
+
+class Filtering(BaseModule):
+    """
+    This class can be used to filter out the data where:
+     - Some samples are not included in the metadata.
+     - Some rows contain non relevant information
+
+    In additon, it can be used to only get the desired rows to study.
+    This class assumes that the samples are placed on the different
+    columns and features (or taxa) are specified on the rows. Ex:
+    samples  1  2  3  4
+    feature1 x  y  t  g
+    feature2 f  h  j  k
+    feature3 l  m  o  p
+    self.steps can be used to save the different argurments used to
+    select the desired data.
+    """
+    def __init__(self, dataframe):
```

Comment:
Then if inheriting from `BaseFiltering` you could use super in the __init__ and simply add a `self.steps` attribute.